opensimplexnoise: Fix persistence, tweak documentation URL and layout

pseidemann opened the pull request:
persistence's spinbox step size was set to 1, breaking the default of 0.5.

changed step size to 0.05 because small changes in this value make a big different (only values 0 to ~2 are relevant).

other fixes I did:

- typo persistance -> persistence
- documentation button led to 404, changed to permanent 3.6 url (I think this is fair since the demo is made with this version and future godot versions might work differently)
- decreased spinbox widths
- increased button sizes to include "3.6" in the doc button name
- increased vbox separation
- centered texture
- ~disabled high dpi since it doesn't actually properly scale on high dpi~

before
<img width="624" alt="Bildschirmfoto 2022-11-18 um 10 58 55" src="https://user-images.githubusercontent.com/431162/202677276-7f3880e6-3ff8-4ecb-9da8-d639834ef4df.png">

after
<img width="1136" alt="Bildschirmfoto 2022-11-18 um 10 59 36" src="https://user-images.githubusercontent.com/431162/202677294-24e78612-6ae9-4973-9a5c-64dd365fc37f.png">

ps: how is the release on the website handled?

Calinou commented:
> disabled high dpi since it doesn't actually properly scale on high dpi

Please look into adding support for multiple resolutions instead. We shouldn't disable hiDPI support as operating systems' lowDPI fallbacks behave badly in games.

pseidemann commented:
hi @Calinou,
I was confused for a second, because as far as I know, there is no proper way to support high dpi with godot.

I read the article you linked and indeed it says this:

> Note:
> Godot doesn't support manually overriding the 2D scale factor yet, so it is not possible to have hiDPI support in non-game applications. Due to this, it is recommended to leave Allow Hidpi disabled in non-game applications to allow for the OS to use its low-DPI fallback.

this is a non-game application, as far as I can tell, since it's only using control nodes. so disabling the flag like I did is following the guide, if I'm not mistaken.

further, it states
> By default, Godot projects aren't considered DPI-aware by the operating system.

but also
> The Godot editor itself is always marked as DPI-aware.

how does the godot editor support high dpi?

how should this be handled if I have a game and I render controls on top of it?

pseidemann commented:
to be complete, following this guide, we should also set stretch mode to disabled and stretch aspect to ignore.
and indeed, right now, when I resize the window, controls will be scaled, which is bad since they can get very small/big, making it unreadable or harder to use.

Calinou commented:
> this is a non-game application, as far as I can tell, since it's only using control nodes. so disabling the flag like I did is following the guide, if I'm not mistaken.

You can use the `2d` stretch mode here, as many other GUI-only demo projects are already doing. It's not a huge problem for demo projects if text becomes too large at high resolutions, as long as the text remains readable on high DPI screens.

pseidemann commented:
hi @Calinou,
allow hidpi + 2d stretch mode as it was before doesn't work for me.
the gui is too small and almost not readable on my system. I'm on mac os with a 27" 4k display (3840x2160 native) with 2304x1296 scaled resolution.

screenshot with godot website for size reference:
<img width="1166" alt="Bildschirmfoto 2022-11-19 um 14 29 19" src="https://user-images.githubusercontent.com/431162/202853107-021b11f9-b741-4336-a86e-ef2d269e5852.png">

however, with the recommended settings from the guide which are just the defaults, the gui size is correct:
<img width="1709" alt="Bildschirmfoto 2022-11-19 um 14 30 57" src="https://user-images.githubusercontent.com/431162/202853154-40103e5f-d2e4-4fd0-a635-146cabc05b49.png">

am I missing something?

Calinou commented:
> allow hidpi + 2d stretch mode as it was before doesn't work for me.

You need to manually resize the window to make it larger on hiDPI displays, as Godot doesn't do this automatically if you've enabled **Allow hiDPI**.

pseidemann commented:
hi @Calinou,

> You need to manually resize the window to make it larger on hiDPI displays [...]

sorry, how is this better than just using the defaults? then no resize would be needed.

> [...] it often messes up with fullscreen applications (especially on Windows).

isn't this a separate bug and should be fixed within the engine?

> [...] Godot should be made to automatically change the initial window size [...]

hmm this doesn't make sense for me. I, as a user of a gui application, should be able to choose a arbitrary window size, by resizing the window to my liking (of course their might be a minimum size but this doesn't matter here) and the gui elements should always have the same size, no matter how big or small I want to make the window.

I'm fine with changing it back to "allow hidpi", also for consistency sake with other demos, but I'm interested in your reasoning, because it seems like we have very different understandings how gui applications should behave. I'm just comparing here with other existing applications like blender or even the godot editor itself. they behave very differently from what you describe as a desired behavior.

Calinou commented:
> isn't this a separate bug and should be fixed within the engine?

No, this is a Windows bug that was never fixed and will probably never be. Plenty of AAA games still run into this issue nowadays because their executable doesn't mark the game window as DPI-aware.

pseidemann commented:
independent of the new convention in 4.0, don't you think there is a kind of surprising discrepancy?

pseidemann commented:
reverted the allow hidpi change and moved the discussion regarding scaling to https://github.com/godotengine/godot/issues/42085.

good to merge from my side

pseidemann commented:
cc @Calinou

pseidemann commented:
hi @Calinou could you merge this please?
thanks! :smiley:

Calinou reviewed (approved):
Tested locally, it works. Thanks!

Calinou commented:
Please check if some of the updates are relevant for the `4.0-dev` branch's Noise Viewer, which uses FastNoiseLite (it replaces OpenSimplexNoise in 4.0): https://github.com/godotengine/godot-demo-projects/tree/4.0-dev/misc/noise_viewer
